packages/block-editor/src/hooks/border-style.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * WordPress dependencies
+ */
+import { getBlockSupport } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import BorderStyleControl from '../components/border-style-control';
+import useEditorFeature from '../components/use-editor-feature';
+import { BORDER_SUPPORT_KEY } from './border';
+import { cleanEmptyObject } from './utils';
+
+/**
+ * Inspector control for configuring border style property.
+ *
+ * @param  {Object} props  Block properties.
+ * @return {WPElement}     Border style edit element.
+ */
+export const BorderStyleEdit = ( props ) => {
+	const {
+		attributes: { style },
+		setAttributes,
+	} = props;
+
+	if ( useIsBorderStyleDisabled( props ) ) {
+		return null;
+	}
+
+	const onChange = ( newBorderStyle ) => {
+		const newStyleAttributes = {
```

Comment:
here `merge` can be used as well

Reply:
It could be. I'll make this change if it is [decided we want to increase lodash usage](https://github.com/WordPress/gutenberg/pull/30124#discussion_r600094420).